### PR TITLE
Adds aria-expanded for See more options.

### PR DESCRIPTION
**Pull request by justinlittman**
refs #3170

# Why was this change made? 🤔

# How was this change tested? 🤨

⚡ ⚠ If this change involves consuming from or writing to another service (or shared file system), ***run [integration test create_object_h2_spec.rb](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test manually in [stage|qa] environment, in addition to specs. ⚡

# Does your change introduce accessibility violations? 🩺

⚡ ⚠ Please ensure this change does not introduce accessibility violations (at the WCAG A or AA conformance levels); if it does, include a rationale. See the [Infrastructure accessibility guide](https://github.com/sul-dlss/DeveloperPlaybook/blob/main/best-practices/infra-accessibility.md) for more detail. ⚡

### CODE REVIEW

File: app/components/works/subtypes_component.html.erb [hunk 1]
```diff
@@ -39,7 +39,7 @@
     </div>
 
     <% unless mixed_material_type? %>
-      <a href="#" class="more-options collapsed" data-action="more-types#toggleMoreTypes" data-more-types-target="moreTypesLink">
+      <a href="#" class="more-options collapsed" data-action="more-types#toggleMoreTypes" data-more-types-target="moreTypesLink" aria-expanded="false" role="button">
```
Comment (edsu) [inline app/components/works/subtypes_component.html.erb:42]: Does this need a `aria-controls` attribute as well?

File: app/components/works/work_type_modal_component.html.erb [hunk 2]
```diff
@@ -65,7 +65,7 @@
 
           <div class="mb-4 subtype-container" data-work-type-modal-target="subtype"></div>
 
-          <a href="#" class="more-options collapsed" data-action="work-type-modal#toggleMoreTypes" data-work-type-modal-target="moreTypesLink" hidden>
+          <a href="#" class="more-options collapsed" data-action="work-type-modal#toggleMoreTypes" data-work-type-modal-target="moreTypesLink" hidden aria-expanded="false" role="button">
```
Comment (edsu) [inline app/components/works/work_type_modal_component.html.erb:68]: same question here re: `aria-controls`

Comment (justinlittman) [reply app/components/works/work_type_modal_component.html.erb]: Added.

File: app/components/works/subtypes_component.html.erb (hunk 1 above)
Comment (justinlittman) [reply app/components/works/subtypes_component.html.erb]: Added.

**Review by edsu — APPROVED** [review]

File: app/javascript/controllers/work_type_modal_controller.js [hunk 3]
```diff
@@ -61,13 +61,15 @@ export default class extends Controller {
...
     this.moreTypesLinkTarget.classList.toggle('collapsed', false)
+    this.moreTypesLinkTarget.setAttribute('aria-expanded', true)
     this.continueButtonTarget.focus()
   }
 
   hideMoreTypes() {
     this.moreTypesTarget.hidden = true
     this.moreTypesLinkTarget.innerHTML = 'See more options'
     this.moreTypesLinkTarget.classList.toggle('collapsed', true)
+    this.moreTypesLinkTarget.setAttribute('aria-expanded', false)
```
Comment (edsu) [inline app/javascript/controllers/work_type_modal_controller.js:72]: Not a blocker, but do we need `aria-controls` here too?

**Pull request merged**